packages/discovery-provider/src/tasks/entity_manager/entities/user.py

```diff
@@ -535,12 +535,33 @@ def update_user_associated_wallets(
                     blocknumber=user_record.blocknumber,
                     blockhash=user_record.blockhash,
                 )
-                added_wallets.append(associated_wallet_entry)
-        is_updated_wallets = set(
-            [prev_wallet.wallet for prev_wallet in previous_wallets]
-        ) != set([wallet.wallet for wallet in added_wallets])
-
-        if is_updated_wallets:
+                current_wallets.append(associated_wallet_entry)
+
+        # Create wallet address sets for each list
+        previous_wallets_set = set([wallet.wallet for wallet in previous_wallets])
+        current_wallets_set = set()
+        for wallet in current_wallets:
+            # Check and throw if current_wallets has duplicate wallet addresses
+            if wallet.wallet in current_wallets_set:
+                raise Exception("Duplicate wallet in list of current wallets")
```

Comment:
What happens when we throw here? Does indexing halt? Should this throw or just ignore duplicates?

Reply:
this should throw and hopefully get caught in the except block below so that we skip this update but continue indexing

Reply:
Does this mean this user is unable to update their associated wallets due to being in a bad state? or how would they get out of the state where `current_wallets` has a dupe? Would it not make more sense to gracefully handle this with the sets?

Reply:
previous wallets is the state in the db and current wallets is the state that the user is trying to update to. Maybe not the best naming. From what i understand, there should be no way for the db state to have duplicate wallets and this is just a check to make sure that the user is not trying to update to a state with multiple wallets, but the client should prevent this from happening before we ever get here. This is just a fix for a bug that has been existing here so I don't think that it will have much of an effect

Reply:
Ah gotcha, thanks! I misunderstood "current" to be "before update"